client/src/routes/Quiz.tsx

```diff
@@ -0,0 +1,95 @@
+        setSelectedOption(optionId);
+        const updatedQuiz = { ...quiz };
+        updatedQuiz.questions[currentQuestion].selected_option_id = optionId;
+
+        // If you need to do something with updatedQuiz, like saving it, you can handle that here
+    };
+    const isPrevDisabled = () => {
+        return currentQuestion === 0;
+    };
+    const isNextDisabled = () => {
+        return (
+            currentQuestion === quiz.questions.length - 1 &&
+            selectedOption === ""
+        );
+    };
+
+    return (
+        <div className="container items-center flex flex-col py-20">
+            <div>
+                <h1>{quiz.questions[currentQuestion].text}</h1>
+            </div>
+            {quiz.questions[currentQuestion] && (
+                <div>
+                    <ul>
+                        {quiz.questions[currentQuestion].options.map(
+                            (option) => (
+                                <li key={option.id}>
+                                    <label>
+                                        <input
+                                            type="radio"
+                                            name="option"
+                                            value={option.id}
+                                            checked={
+                                                selectedOption === option.id
+                                            }
+                                            onChange={() =>
+                                                handleOptionChange(option.id)
+                                            }
+                                        />
+                                        {option.text}
+                                    </label>
+                                </li>
+                            ),
+                        )}
+                    </ul>
+                </div>
+            )}
+            <button
+                style={{
+                    position: "fixed",
+                    bottom: "20px",
+                    left: "20px",
+                    width: "100px",
+                    borderRadius: "4px",
+                    color: "white",
+                    backgroundColor: "darkred",
+                    height: "40px",
+                }}
+                disabled={isPrevDisabled()}
+                onClick={() => setCurrentQuestion(currentQuestion - 1)}
+            >
+                Previous
+            </button>
+            <button
+                style={{
+                    position: "fixed",
+                    bottom: "20px",
+                    right: "20px",
+                    width: "100px",
+                    borderRadius: "4px",
+                    color: "white",
+                    backgroundColor: "green",
+                    height: "40px",
+                }}
+                disabled={isNextDisabled()}
+                onClick={() => setCurrentQuestion(currentQuestion + 1)}
```

Comment:
its a pattern to use it like 
`setCurrentQuestion(currentQuestion => currentQuestion + 1)`